src/cr/cube/matrix/measure.py

```diff
@@ -1320,6 +1325,118 @@ def blocks(self):
+        """
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.blocks`" % type(self).__name__
+        )
+
+    @lazyproperty
+    def direction(self):
+        """String, either "row" or "column", the directionality the marginal."""
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.direction`" % type(self).__name__
+        )
+
+    @lazyproperty
+    def is_defined(self):
+        """Bool, indicating whether the marginal can be calculated.
+
+        Defaults to True, but designed to be overridden by subclasses if necessary.
+        """
+        return True
+
+
+class _BaseScaledCountMarginal(_BaseSecondOrderMarginal):
+    """A base class for marginals that depend on the scaled counts."""
+
+    @lazyproperty
+    def _columns_have_numeric_values(self):
+        """Bool indicating whether any numeric values on columns dimension are defined."""
+        return not np.all(np.isnan(self._columns_numeric_values))
+
+    @lazyproperty
+    def _columns_numeric_values(self):
+        """ndarray of numeric_values from the columns dimension"""
+        return np.array(self._dimensions[1].numeric_values)
+
+    def _weighted_median(self, values, counts, axis):
+        """ndarray of float64 the median of values, weighted by counts along axis"""
+        # --- TODO: This should be rewritten so that it doesn't use `np.repeat` because
+        # --- it's inefficient to make the whole array, and also because it doesn't
+        # --- correctly handle fractional counts that could come with weights. Example
+        # --- implementation:
+        # --- https://github.com/nudomarinero/wquantiles/blob/master/wquantiles.py
+        not_a_nan_index = ~np.isnan(values)
+        values = values[not_a_nan_index]
+        counts = np.nan_to_num(counts[:, not_a_nan_index]).astype("int64")
+        scale_median = np.array(
+            [
+                self._median(np.repeat(values, counts[i, :]))
+                for i in range(counts.shape[0])
+            ]
+        )
+        return scale_median
+
+    def _median(self, values):
+        """float64 median of values but without warning when there are no values"""
+        return np.median(values) if values.size != 0 else np.nan
+
+
+class _RowsScaleMedian(_BaseScaledCountMarginal):
+    """Provides the rows scale median marginals for a matrix if available.
+
+    The rows scale median is a 1D np.float64 ndarray, the same dimension as a column, that
+    has the weighted median value of column numeric values.
+
+    The rows scale median is not defined when no columns have numeric values defined.
+    """
+
+    @lazyproperty
+    def blocks(self):
+        """List of the 2 1D ndarray "blocks" of the rows-scale-median.
+
+        These are the base-values and the subtotals.
+        """
+        if not self.is_defined:
+            raise ValueError("No numeric values are defined on the columns dimension.")
```

Comment:
I'm thinking we should probably stick to speaking in caller-space for an exception like this one, meaning something like:

"rows-scale-median is undefined when no numeric values are defined on columns dimension".

In this case, no caller on `_Slice` is going to see this today, but still, might as well spell out as much as possible, like "this exception was raised because it is undefined for this value (cube-result)" and "the reason this cube-result is insufficient is that it has no numeric values on the columns dimension". This gives the caller all they need to know to fix the problem.